examples/gno.land/r/docs/minisocial/v3/dao.gno

```diff
@@ -0,0 +1,74 @@
+package minisocial
+
+import (
+	"github.com/gnolang/gno/gnovm/stdlibs/std"
+	"gno.land/p/nt/commondao"
+	cmndao "gno.land/r/nt/commondao"
+)
+
+var DAO = cmndao.New("My DAO")
+
+func init() {
+	DAO.Members().Add("g125em6arxsnj49vx35f0n0z34putv5ty3376fg5") // @leohhhn
+}
+
+// RemovePostProp implements ProposalDefinition
+type RemovePostProp struct {
+	postToRemoveID string
+	reason         string
+}
+
+func (RemovePostProp) VotingPeriod() time.Duration { return time.Hour * 24 * 7 }
+
+func (p RemovePostProp) Title() string {
+	return "Proposal to remove post " + p.postToRemoveID
+}
+
+func (p RemovePostProp) Body() string {
+	return "We'd like to remove the post because " + p.reason
+}
+
+// what if i want some specific context that I
+// hardcode at the time of the proposal? ie:
+// i want to propose a specific balance/budget exactly at the price when
+// i propose something... -> Take a look at govdao v3 "Metadata"
+// the issue is that Execute will be called at some point,
+// with some state, which has advanced since I made the proposal
+// I should be able to do either:
+// execute prop at time when its voted yes (ie coin*amt == 10$)
+// or, record price at time of exec, get val of coin, send exactly 10$
+// at the time of execution of the prop
```

Comment:
Wouldn't it be possible to create the `RemovePostProp` instance assigning any value you need to check when executing? The metadata would be represented as properties of the proposal definition, in this case properties of `RemovePostProp`.

Usually other implementations, like gno.me which uses a similar approach for custom proposal types, uses properties in the proposal definition.

Reply:
> -> Take a look at govdao v3 "Metadata"

The metadata is the proposal definition itself, you could use one or more properties to save any information or reference that you need to be available during execution, validation or tallying.